skbio/sequence/_sequence.py

```diff
@@ -1467,11 +1461,8 @@ def replace(self, where, character):
         index = self._munge_to_index_array(where)
         seq_bytes = self._bytes.copy()
         seq_bytes[index] = character
-        metadata = self.metadata if self.has_metadata() else None
-        positional_metadata = self.positional_metadata if \
-            self.has_positional_metadata() else None
-        return self.__class__(seq_bytes, metadata=metadata,
-                              positional_metadata=positional_metadata)
+        return self.__class__(seq_bytes, metadata=self.metadata,
```

Comment:
It's `__class__` to run validations, since the user could introduce invalid characters. I'll add a comment